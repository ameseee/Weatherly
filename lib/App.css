
body {
  align-items: center;
  /*background-image: url('http://imgur.com/a/lKS2R');*/
  background-image: url('http://www.wallcoo.net/cartoon/abstract_colors_2560x1600_0902/images/abstract_colors_Destiny-Islands---morning.jpg');
  background-repeat: no-repeat;
  background-size: cover;
  display: flex;
  flex-direction: column;
  height: 100vh;
  width: 100%;
  justify-content: center;
}
