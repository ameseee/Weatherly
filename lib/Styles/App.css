#app {
  width: 100%;
}

body {
  align-items: center;
  /*background-image: url('http://imgur.com/a/lKS2R');*/
  /*background-image: url('http://i.imgur.com/AgoEHWA.png');*/
  /*background-repeat: no-repeat;*/
  /*background-size: cover;*/
  color: #fff;
  background-color: #029bf1;
  display: flex;
  flex-direction: column;
  height: 100vh;
  width: 100%;
  justify-content: center;
}
