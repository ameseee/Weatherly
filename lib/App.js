import React from 'react';
import './App.css';
import Search from './Search';
import CurrentWeather from './CurrentWeather';
import SevenHour from './SevenHour';
import TenDay from './TenDay';
import Key from './Key'

// const weatherInfo = (state, city) => {
//   `http://api.wunderground.com/api/key/conditions/q/{state}/{user input city}.json`
// }

export default class App extends React.Component {

  constructor(props) {
    super(props)
    this.state = {weatherData: []}
  }

  componentDidMount() {
    fetch(`http://api.wunderground.com/api/${Key}/conditions/q/CO/${city}.json`)
    .then(data => data.json())
    .then(data => {
      this.setState({
        weatherData: data
      })
    })
  }


  render() {
console.log('hi');
    if (!this.state.weatherData) {
      return <p>loading...</p>
    }

    return (
      <div>
         <Search
         city={this.search}/>
        {/*<CurrentWeather />*/}
        {/*<SevenHour />
        <TenDay />*/}
      </div>
    );

  }
}

{/*weatherData={this.state.weatherData} THIS BELONGS IN SEARCH WITHIN RENDER() */}
