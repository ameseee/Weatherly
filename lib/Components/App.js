import React from 'react';
import '../Styles/App.css';
import Welcome from './Welcome';
import CurrentWeather from './CurrentWeather';
import SevenHour from './SevenHour';
import TenDay from './TenDay';
import HourlyCard from './HourlyCard';
import DailyCard from './DailyCard';
import API from '../utils/Api';

export default class App extends React.Component {

  constructor(props) {
    super(props)
    this.state = {
                  ready: false,
                  weatherData: {},
                  }
  }

  componentDidMount() {
    if (localStorage.getItem('location')) {
      var parsedLocal = localStorage.getItem('location');
      this.getCityWeather(parsedLocal)
    }
  }

  getCityWeather(location) {

    localStorage.setItem('location', location);
    fetch(API(location))
    .then(data => data.json())
    .then(data => {
      console.log('made api call');
      this.setState({
        ready: true,
        weatherData: {
          place: data.current_observation.display_location.full,
          currentCondition: data.current_observation.weather,
          currentDay: data.forecast.simpleforecast.forecastday[0].date.weekday,

          currentDate: {
                        month: data.forecast.simpleforecast.forecastday[0].date.month,
                        day: data.forecast.simpleforecast.forecastday[0].date.day,
                        year: data.forecast.simpleforecast.forecastday[0].date.year
                        },

          currentTemp: data.current_observation.temp_f,
          dailyHighTemp: data.forecast.simpleforecast.forecastday[0].high.fahrenheit,
          dailyLowTemp: data.forecast.simpleforecast.forecastday[0].low.fahrenheit,
          summary: data.forecast.txt_forecast.forecastday[0].fcttext,
          hourlyForecast: data.hourly_forecast,

          dailyForecast: data.forecast.simpleforecast.forecastday
        }
      })
    }).catch( () => {alert("Please enter valid City and State")})
  }

  render() {

    const {place, currentCondition, currentDay, currentTemp, dailyHighTemp, dailyLowTemp, summary, hourlyForecast, dailyForecast} = this.state.weatherData;


    const hourCards = (hourlyForecast || []).map( (hour, i) => {
      if (i < 7) {
        return <HourlyCard temp={hour.temp.english} time={hour.FCTTIME.civil} img={hour.icon_url} key={hour.FCTTIME.civil} />
      }
    })

    const dayCards = (dailyForecast || []).map( (day, i) => {
      if (i < 10) {
        return <DailyCard key={day.period} highT={day.high.fahrenheit} lowT={day.low.fahrenheit} day={day.date.weekday} img={day.icon_url} />
      }
    })

    return (
      <div>

        {(!this.state.ready) &&
          <Welcome
          getApi={this.getCityWeather.bind(this)}/>}

        {(this.state.ready) &&
          <div>
            <CurrentWeather
            getNewLocation={this.getCityWeather.bind(this)}
            getApi={this.getCityWeather.bind(this)}
            place={place}
            condition={currentCondition}
            day={currentDay}
            dateMonth={this.state.weatherData.currentDate.month}
            dateDay={this.state.weatherData.currentDate.day}
            dateYear={this.state.weatherData.currentDate.year}
            temp={currentTemp}
            highT={dailyHighTemp}
            lowT={dailyLowTemp}
            summary={summary} />

            <SevenHour>
              {hourCards}
            </SevenHour>

            <TenDay>
              {dayCards}
            </TenDay>
          </div>}

      </div>
    )
  }
}
