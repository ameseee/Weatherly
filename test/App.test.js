import React from 'react';
import { shallow, mount } from 'enzyme';
import App from '../lib/Components/App';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />)
  })

  // afterEach(() => {
  //   localStorage.clear()
  // })

  it('should exist', () => {
    expect(wrapper).toBeDefined()
  })

  // it.skip('should render the Header and ToDontList component', () => {
  // expect(wrapper.find("Header").length).toEqual(1)
  // expect(wrapper.find("Header")).toBeDefined()
  //
  // expect(wrapper.find("ToDontList").length).toEqual(1)
  // expect(wrapper.find("ToDontList")).toBeDefined()
  // })

  it('should render the Welcome component', () => {
    expect(wrapper.find("Welcome").length).toEqual(1)
    expect(wrapper.find("Welcome")).toBeDefined()
  })

// TODO: I  think in these other render test below Ill need to make the this.state.ready to be true

  it.skip('should render the CurrentWeather component', () => {
    expect(wrapper.find("CurrentWeather").length).toEqual(1)
    expect(wrapper.find("CurrentWeather")).toBeDefined()
  })

  it.skip('should render the SevenHour component', () => {
    expect(wrapper.find("SevenHour").length).toEqual(1)
    expect(wrapper.find("SevenHour")).toBeDefined()
  })

  it.skip('should render the TenDay component', () => {
    expect(wrapper.find("TenDay").length).toEqual(1)
    expect(wrapper.find("TenDay")).toBeDefined()
  })

  it.skip('initially should have a state of toDonts set to an empty array', () => {
    expect(wrapper.state()).toEqual({ toDonts: [] })
    expect(wrapper.state().toDonts).toEqual(expect.arrayContaining([]))
  })

  it.skip('should retrieve data from local storage on mount', () => {
    // const toDonts = [
    //   {title: 'title', body: 'body', id: 1},
    //   {title: 'title', body: 'body', id: 2}
    // ]

    localStorage.setItem('toDonts', JSON.stringify(toDonts))

    wrapper = mount(<App />)
    expect(wrapper.state().toDonts).toEqual(toDonts)
  })

  it.skip('should add toDonts to state', () => {
    expect(wrapper.state().toDonts.length).toEqual(0)

    // const newToDonts = [
    //   { title: 'dumpster food', body: 'dont eat out of the trash', id: 1 }
    //   { title: 'wiz kid', body: 'dont pee on the sidewalk', id: 2 }
    // ]
    wrapper.instance().addToDont(newToDonts)

    const itemsInStorage = JSON.parse(localStorage.getItem('toDonts')).length

    expect(wrapper.state(toDonts).length).toEqual(2)
    expect(wrapper.state().toDonts[0]).toEqual(newToDont)
    expect(itemsInStorage).toEqual(2)
  })

  it.skip('should update state with pre-existing objects', () => {
    wrapper = shallow(<App />)

    const toDonts = [
      { title: 'title', body: 'body', id: 1 }
    ]

    wrapper.setState({toDonts: toDonts})

    expect(wrapper.state().toDonts.length).toEqual(1)

    const newToDont ={ title: 'title', body: 'body', id: 2 }

    expect(wrapper.state().toDonts.length).toEqual(2)
  })


})
