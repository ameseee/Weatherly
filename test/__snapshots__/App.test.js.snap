// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the CurrentWeather component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <Welcome
        getApi={[Function]}
    />
</div>,
  "nodes": Array [
    <div>
      <Welcome
            getApi={[Function]}
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <App />,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": App {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "ready": false,
          "weatherData": Object {},
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <Welcome
                    getApi={[Function]}
          />
</div>,
        "_debugID": 6,
        "_renderedOutput": <div>
          <Welcome
                    getApi={[Function]}
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;

exports[`App should render the SevenHour component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <Welcome
        getApi={[Function]}
    />
</div>,
  "nodes": Array [
    <div>
      <Welcome
            getApi={[Function]}
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <App />,
      "_debugID": 7,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": App {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "ready": false,
          "weatherData": Object {},
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 4,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <Welcome
                    getApi={[Function]}
          />
</div>,
        "_debugID": 8,
        "_renderedOutput": <div>
          <Welcome
                    getApi={[Function]}
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;

exports[`App should render the TenDay component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <Welcome
        getApi={[Function]}
    />
</div>,
  "nodes": Array [
    <div>
      <Welcome
            getApi={[Function]}
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <App />,
      "_debugID": 9,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": App {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "ready": false,
          "weatherData": Object {},
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 5,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <Welcome
                    getApi={[Function]}
          />
</div>,
        "_debugID": 10,
        "_renderedOutput": <div>
          <Welcome
                    getApi={[Function]}
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;
